### PR TITLE
Add unique css handle to each stack item

### PR DESCRIPTION
**Pull request by Klynger**
With the current behavior, there is no way to position each element in the stack, with this handle you can do it now.

See the preview button in this [workspace](https://quickview--storecomponents.myvtex.com/)

image

### Extra information

Since the stack will never change position because is passed by the store theme the index will always be the same for all the component lifecycle

### CODE REVIEW

File: react/index.tsx [hunk 1]
```diff
@@ -2,13 +2,16 @@ import React from 'react'
 import { defineMessages } from 'react-intl'
 import { useCssHandles, applyModifiers } from 'vtex.css-handles'
 
-const CSS_HANDLES = ['stackContainer', 'stackItem'] as const
+const CSS_HANDLES = ['stackContainer', 'stackItem', 'item'] as const
```
Comment (victorhmp) [inline react/index.tsx]:
```suggestion
const CSS_HANDLES = ['stackContainer', 'stackItem'] as const
```

**Review by victorhmp — CHANGES_REQUESTED** [review]

**Review by victorhmp — APPROVED** [review]

**Review by iaronaraujo — APPROVED** [review]

**Review by klzns — APPROVED** [review]

File: react/index.tsx [hunk 2]
```diff
@@ -8,16 +8,26 @@ interface Props {
...
+  children,
+  zIndexOffset = 0,
+}) => {
   const handles = useCssHandles(CSS_HANDLES)
   return (
     <div className={`${handles.stackContainer} relative`}>
       {React.Children.toArray(children).map((child, idx) => {
+
+        const childBlockClass = (child as any)?.props?.blockProps?.blockClass ? (child as any).props.blockProps.blockClass : ''
+        let stackItemExtraBlockClass = ''
```
Comment (lbebber) [inline react/index.tsx]: ⛏ acho q da pra tirar o "extra" do nome

File: react/index.tsx [hunk 3]
```diff
@@ -8,16 +8,26 @@ interface Props {
...
+const StackLayout: StorefrontFunctionComponent<Props> = ({
+  children,
+  zIndexOffset = 0,
+}) => {
   const handles = useCssHandles(CSS_HANDLES)
   return (
     <div className={`${handles.stackContainer} relative`}>
       {React.Children.toArray(children).map((child, idx) => {
+
+        const childBlockClass = (child as any)?.props?.blockProps?.blockClass ? (child as any).props.blockProps.blockClass : ''
```
Comment (lbebber) [inline react/index.tsx]: ⛏ isso nao rola talvez? (nunca usei o `?`, nao tenho certeza):

`(child as any)?.props?.blockProps?.blockClass || '')`

Comment (lbebber) [inline react/index.tsx]:
```suggestion
        // This allows the user customize the stackItem via CSS, via the blockClass of each child
        const childBlockClass = (child as any)?.props?.blockProps?.blockClass ? (child as any).props.blockProps.blockClass : ''
```

**Review by lbebber — COMMENTED** [review]
umas sugestoezinhas, mas parece bom

acho que isso tb requer atualizar documentaçao

**Review by lbebber — APPROVED** [review]

**Pull request merged**